src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs

```diff
@@ -0,0 +1,340 @@
+﻿using Uno.Extensions;
+using Uno.UI.DataBinding;
+using Windows.UI.Xaml;
+using Uno.UI.Extensions;
+using Windows.UI.Xaml.Data;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using Uno.Disposables;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Linq;
+using Foundation;
+#if __MACOS__
+using AppKit;
+using _NativeScrollView = AppKit.NSScrollView;
+using _View = AppKit.NSView;
+#else
+using UIKit;
+using _NativeScrollView = UIKit.UIScrollView;
+using _View = UIKit.UIView;
+#endif
+using CoreGraphics;
+using Uno.Logging;
+using Windows.Foundation;
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class ScrollContentPresenter : _NativeScrollView, DependencyObject
+	{
+		private CGSize _measuredSize;
+		private bool _requiresMeasure;
+			   
+#if __IOS__
+		public override
+#else
+		public
+#endif
+			CGSize SizeThatFits(CGSize size)
+		{
+			if (_content != null)
+			{
+				var frameworkElement = _content as IFrameworkElement;
+
+				var horizontalMargin = frameworkElement.Margin.Left + frameworkElement.Margin.Right;
+				var verticalMargin = frameworkElement.Margin.Top + frameworkElement.Margin.Bottom;
+
+				size = AdjustSize(size);
+
+
+				var availableSizeForChild = size;
+				if (!(_content is IFrameworkElement))
+				{
+					// Apply margin if the content is native (otherwise it will apply it itself)
+					availableSizeForChild.Width -= (nfloat)horizontalMargin;
+					availableSizeForChild.Height -= (nfloat)verticalMargin;
+				}
+
+#if __IOS__
+				_measuredSize = _content.SizeThatFits(availableSizeForChild);
+#else
+				_measuredSize = (_content as FrameworkElement)?.SizeThatFits(availableSizeForChild) ?? default(CGSize);
+#endif
+				_measuredSize.Width += (nfloat)horizontalMargin;
+				_measuredSize.Height += (nfloat)verticalMargin;
```

Comment:
### [Codacy](https://app.codacy.com/gh/unoplatform/uno/commit?cid=437596529) detected an issue:
#### Message: `'_measuredSize' is null on at least one execution path.`

    
#### Currently on:
+ **Commit**:  934d3d5ec6c416105400c8b388d95495d64afa33
+ **File**:    [src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs](https://github.com/unoplatform/uno/blob/934d3d5ec6c416105400c8b388d95495d64afa33/src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs)
+ **LineNum**: [64](https://github.com/unoplatform/uno/blob/934d3d5ec6c416105400c8b388d95495d64afa33/src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs#L64)